packages/devtools_app/lib/src/screens/memory/panes/diff/model.dart

```diff
@@ -2,45 +2,43 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-import 'package:flutter/widgets.dart';
+import 'package:flutter/foundation.dart';
 import 'package:vm_service/vm_service.dart';
 
-abstract class DiffListItem with ChangeNotifier {
-  String get name;
-
+abstract class DiffListItem {
   /// Number, that, if shown in name, should be unique in the list.
   ///
   /// If the number is not shown, it should be 0.
-  int get nameNumber;
+  int get displayNumber;
 
-  bool isProcessing = false;
+  final _isProcessing = ValueNotifier<bool>(false);
+  ValueListenable<bool> get isProcessing => _isProcessing;
 }
 
 class InformationListItem extends DiffListItem {
-  InformationListItem();
-
-  @override
-  String get name => 'Snapshots';
-
   @override
-  int get nameNumber => 0;
+  int get displayNumber => 0;
 }
 
 class SnapshotListItem extends DiffListItem {
-  SnapshotListItem(this.graph, this.nameNumber, this.isolateName) {
-    isProcessing = true;
-    graph.whenComplete(() {
-      isProcessing = false;
-      notifyListeners();
+  SnapshotListItem(
+    Future<HeapSnapshotGraph?> graphReceiver,
+    this.displayNumber,
+    this._isolateName,
+  ) {
+    _isProcessing.value = true;
+    graphReceiver.whenComplete(() async {
+      _isProcessing.value = false;
+      graph = await graphReceiver;
```

Comment:
should the order of these be switched? is `_isProcessing` really false before `await graphReceiver` completes?

Reply:
It actually already completed and this await will return momentarily, but, yes, it looks strange. Switched.